lib/workers/repository/update/branch/auto-replace.spec.ts

```diff
@@ -200,76 +200,656 @@ describe('workers/repository/update/branch/auto-replace', () => {
       expect(res).toBe(dockerfile.replace(upgrade.depName, upgrade.newName));
     });
 
-    it('handles already replaced', async () => {
-      const dockerfile = 'FROM library/ubuntu:20.04';
-      upgrade.manager = 'dockerfile';
-      upgrade.updateType = 'replacement';
-      upgrade.depName = 'library/alpine';
-      upgrade.newName = 'library/ubuntu';
-      upgrade.packageFile = 'Dockerfile';
-      const res = await doReplacementAutoReplace(
-        upgrade,
-        dockerfile,
-        reuseExistingBranch
-      );
-      expect(res).toBe(dockerfile);
-    });
-
     it('updates with terraform replacement', async () => {
       const hcl =
         'module "foo" {\nsource = "github.com/hashicorp/example?ref=v1.0.0"\n}';
       upgrade.manager = 'terraform';
       upgrade.updateType = 'replacement';
       upgrade.depName = 'github.com/hashicorp/example';
       upgrade.newName = 'github.com/hashicorp/new-example';
+      upgrade.currentValue = 'v1.0.0';
+      upgrade.depIndex = 0;
       upgrade.packageFile = 'modules.tf';
-      const res = await doReplacementAutoReplace(
-        upgrade,
-        hcl,
-        reuseExistingBranch
-      );
+      const res = await doAutoReplace(upgrade, hcl, reuseExistingBranch);
       expect(res).toBe(hcl.replace(upgrade.depName, upgrade.newName));
     });
 
-    it('fails with old name in depname', async () => {
-      const yml =
-        'image: "1111111111.dkr.ecr.us-east-1.amazonaws.com/my-repository:1"\n\n';
-      upgrade.manager = 'regex';
-      upgrade.updateType = 'replacement';
-      upgrade.depName =
-        '1111111111.dkr.ecr.us-east-1.amazonaws.com/my-repository';
-      upgrade.currentValue = '1';
-      upgrade.newName =
-        '1111111111.dkr.ecr.us-east-1.amazonaws.com/my-repository';
-      upgrade.depIndex = 0;
-      upgrade.replaceString =
-        'image: "1111111111.dkr.ecr.us-east-1.amazonaws.com/my-repository:1"\n\n';
-      upgrade.packageFile = 'k8s/base/defaults.yaml';
-      upgrade.matchStrings = [
-        'image:\\s*\\\'?\\"?(?<depName>[^:]+):(?<currentValue>[^\\s\\\'\\"]+)\\\'?\\"?\\s*',
-      ];
-      const res = doReplacementAutoReplace(upgrade, yml, reuseExistingBranch);
-      await expect(res).rejects.toThrow(WORKER_FILE_UPDATE_FAILED);
-    });
-
     it('rebases if the deps list has changed', async () => {
       upgrade.baseDeps = extractPackageFile(sampleHtml)?.deps;
       reuseExistingBranch = true;
-      const res = await doReplacementAutoReplace(
+      const res = await doAutoReplace(
         upgrade,
         'existing content',
         reuseExistingBranch
       );
       expect(res).toBeNull();
     });
   });
+
+  describe('doReplacementAutoReplaceNewDepName', () => {
+    let reuseExistingBranch: boolean;
+    let upgrade: BranchUpgradeConfig;
+
+    beforeEach(() => {
+      upgrade = {
+        ...JSON.parse(JSON.stringify(defaultConfig)),
+      };
```

Comment:
```suggestion
      upgrade = getConfig();
```

`getConfig` will always return a fresh config